play/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/GenerateResources.kt

```diff
@@ -50,85 +52,47 @@ internal abstract class GenerateResources : DefaultTask() {
-            }.lastOrNull() // Pick the most specialized option available. E.g. `paidProdRelease`
-
-            val files = parameters.resources.get()
-                    .filterNot { it.isDirectory }
-                    .sortedBy { file ->
-                        val dir = parameters.inputDirs.get().singleOrNull {
-                            file.startsWith(it.asFile)
-                        }
-                        parameters.inputDirs.get().indexOf(dir)
-                    }
-                    .ifEmpty { return }
-
-            val changedDefaults = mutableListOf<File>()
-            for (file in files) {
-                file.validate()
-
-                defaultLocale.nullOrFull()?.let {
-                    if (file.isChildOf(LISTINGS_PATH) && file.isChildOf(it)) {
-                        changedDefaults += file
-                    }
-                }
-                file.copy(file.findClosestDir().findDest())
-            }
-
-            val writeQueue = mutableListOf<Action<Unit>>()
-            for (default in changedDefaults) {
-                val listings = default.findDest().climbUpTo(LISTINGS_PATH)!!
-                val relativePath = default.invariantSeparatorsPath.split("$defaultLocale/").last()
-
-                listings.listFiles()
-                        .orEmpty()
-                        .filter { it.name != defaultLocale }
-                        .map { File(it, relativePath) }
-                        .filterNot(File::exists)
-                        .filterNot(::hasGraphicCategory)
-                        .forEach {
-                            val destName = it.parentFile.toRelativeString(
-                                    parameters.outputDir.get().asFile)
-                            val dest = File(parameters.outputDir.get().asFile, destName)
-
-                            writeQueue += Action { default.copy(dest) }
-                        }
-            }
-            writeQueue.forEach { it.execute(Unit) }
+            for (file in parameters.files.get()) file.validate()
         }
 
         private fun File.validate() {
-            val areRootsValid = isDirectChildOf(PLAY_PATH)
-                    || isChildOf(LISTINGS_PATH)
-                    || isChildOf(RELEASE_NOTES_PATH)
-                    || isChildOf(RELEASE_NAMES_PATH)
-                    || isChildOf(PRODUCTS_PATH)
+            val areRootsValid = name == PLAY_PATH ||
+                    isDirectChildOf(PLAY_PATH) ||
+                    isChildOf(LISTINGS_PATH) ||
+                    isChildOf(RELEASE_NOTES_PATH) ||
+                    isChildOf(RELEASE_NAMES_PATH) ||
+                    isChildOf(PRODUCTS_PATH)
             check(areRootsValid) { "Unknown Play resource file: $this" }
+            check(extension != INDEX_MARKER) { "Resources cannot use the 'index' extension: $this" }
+            check(name != PLAY_PATH || !isChildOf(PLAY_PATH)) {
+                "The file name 'play' is illegal: $this"
```

Comment:
This was not a requirement in 2.6.x. 

In my app, I have a `flavor` called `play` and it was workin all fine until I upgraded to `2.7`. Should this change be mentioned in release notes?

And also, what is the reason really to keep it strict?

Reply:
Since I have `play` as a flavor. The `bootstrap` task also downloads stuff under `play/play` which then fails when I call `publish` task

Reply:
It would break write precedence: https://github.com/Triple-T/gradle-play-publisher/blob/edab9b6dacb1abd5db4472c018bc11b5ebc7700f/play/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/GenerateResources.kt#L441

But that's a bug, fixed in 4331aa614f676b2a839d50a403ef32e2c103685c.